### ISSUE TITLE
Syntax highlighting will mark the whole file as a string when writing a quote inside a triple quote multiline string

### ISSUE THREAD

**Issue by amoriqbal** [godotengine/godot#41403]
**Godot version:**
3.2 and 4.0 (master, 4e52c75a981f3704a01f702ffd14cc626ea32ec7)

**Issue description:**
The script editor marks up code as multi-line string under certain corner cases.

Eg: If you write the following code in the GdScript editor, all the lines following the declaration of `var some_text` will be marked yellow, as if they are all strings.

```
extends Node
var some_text = """Look at the markup of the rest of the file " """
func _ready():
    print(some_text)
```

But if you run the scene this script will work fine without any error, and you will see the following output under the output tab:

> Look at the markup of the rest of the file

Note that the last `"` that should be present at the end of `Look at the markup of the rest of the file "` is missing

**Steps to reproduce:**
Copy-paste the code, given above, into your GdScript editor. Note the markup (colour of lines 3 and 4).
Attach this script to any node and run the scene. Confirm that the script runs without any error and the _ready function produces the output in the output tab.

**Minimal reproduction project:**
[demo.zip](https://github.com/godotengine/godot/files/5104433/demo.zip)

**Comment by Calinou** [comment]
I can confirm this on 3.2.2:

![Syntax highlighting bug](https://user-images.githubusercontent.com/180032/90815878-ac2dc100-e32b-11ea-8079-78fe2f838311.png)

**Comment by dalexeev** [comment]
The bug is still reproducible in `3.2.3.stable.official`.

**Comment by akien-mga** [comment]
That's expected, it was fixed in the `master` branch (4.0).
